src/enzyme_ad/jax/Passes/OptimizeCommunication.cpp

```diff
@@ -2451,16 +2451,24 @@ struct OptimizeCommunicationPass
     auto context = getOperation()->getContext();
     RewritePatternSet patterns(context);
 
-    // ConcatTwoOperandsCommOptimize is not added to this list. pad_like_raise
-    // is almost certainly a better option.
-    if (permute_like_raise > 0)
-      patterns.add<PeriodicConcatSimplify, RotateCommOptimize, WrapCommOptimize,
-                   ExtendCommOptimize>(context,
-                                       PatternBenefit(permute_like_raise));
-
-    if (pad_like_raise > 0)
-      patterns.add<DUSToPadComm, ConcatToPadCommOptimize>(
-          context, PatternBenefit(pad_like_raise));
+    if (periodic_concat > 0)
+      patterns.add<WrapCommOptimize>(context, PatternBenefit(periodic_concat));
+
+    if (rotate_comm > 0)
+      patterns.add<WrapCommOptimize>(context, PatternBenefit(rotate_comm));
+
+    if (wrap_comm > 0)
+      patterns.add<WrapCommOptimize>(context, PatternBenefit(wrap_comm));
+
+    if (extend_comm > 0)
+      patterns.add<ExtendCommOptimize>(context, PatternBenefit(extend_comm));
+
+    if (dus_to_pad_comm > 0)
+      patterns.add<DUSToPadComm>(context, PatternBenefit(dus_to_pad_comm));
+
+    if (concat_to_pad_comm > 0)
+      patterns.add<ConcatToPadCommOptimize>(context,
```

Comment:
add ConcatTwoOperandsCommOptimize as well with a default of 0?

Reply:
done